### DISCUSSION TITLE
Navigation Guards

### DISCUSSION THREAD

**Discussion by gondc** [posva/unplugin-vue-router#111] [Questions and Help] [answered]
Is there a way to add navigation guards? per route and global?

**Comment by posva** [comment]
Apart from beforeEnter, which you could put in a beforeEach most of the time, they are all the same. What are you trying to do that cannot be done?

**Comment by gondc** [comment]
beforeEnter is what I need, but but you say that I should use beforeEach instead?
Where do I add the beforeEach?

**Reply by posva** [reply, marked as answer]
BeforeEnter runs less times. You can still add them with `definePage()` though. 
Otherwise it’s router.beforeEach()c, when you create it
